lib/multiplexer/multiplexer.go

```diff
@@ -191,14 +178,15 @@ func (m *Mux) Serve() error {
 			go m.detectAndForward(conn)
 			continue
 		}
-		if m.isClosed() {
+		if utils.IsUseOfClosedNetworkError(err) {
+			<-m.context.Done()
```

Comment:
Can this deadlock? I'm always a little wary when there's a receive from channel without select. Would it make sense to add some timeout here?